core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -151,13 +159,14 @@ public Task poll(String taskType, String workerId, String domain) {
 				task.setPollCount(task.getPollCount() + 1);
 				executionDAOFacade.updateTask(task);
 				tasks.add(task);
+			} catch (Exception e) {
+				// db operation failed for dequeued message, re-enqueue with a delay
+				logger.warn("DB operation failed for task: {}, postponing task in queue", taskId, e);
+				queueDAO.postpone(queueName, taskId, 0, queueTaskMessagePostponeSeconds);
```

Comment:
It'd be nice to have a metric for this too, which helps in tracing poll failures due to db operations and find out before the clients notice poll delays.

Reply:
Good idea on the metric, will add.